lite-variables.tf

```diff
@@ -0,0 +1,221 @@
+  type        = list(string)
+  default     = []
+}
+
+variable "hpcc_version" {
+  description = "The version of HPCC Systems to install.\nOnly versions in nn.nn.nn format are supported. Default is 'latest'"
+  type        = string
+  validation {
+    condition     = (var.hpcc_version == "latest") || can(regex("^\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}(-rc\\d{1,3})?$", var.hpcc_version))
+    error_message = "Value must be 'latest' OR in nn.nn.nn format and 8.6.0 or higher."
+  }
+  default = "latest"
+}
+
+variable "storage_data_gb" {
+  type        = number
+  description = "REQUIRED.  The amount of storage reserved for data in gigabytes.\nMust be 10 or more.\nIf a storage account is defined (see below) then this value is ignored."
+  validation {
+    condition     = var.storage_data_gb >= 10
+    error_message = "Value must be 10 or more."
+  }
+  default    = 100
+}
+
+variable "storage_lz_gb" {
+  type        = number
+  description = "REQUIRED.  The amount of storage reserved for the landing zone in gigabytes.\nMust be 1 or more.\nIf a storage account is defined (see below) then this value is ignored."
+  validation {
+    condition     = var.storage_lz_gb >= 1
+    error_message = "Value must be 1 or more."
+  }
+  default    = 25
+}
+
+variable "thor_max_jobs" {
+  type        = number
+  description = "REQUIRED.  The maximum number of simultaneous Thor jobs allowed.\nMust be 1 or more."
+  validation {
+    condition     = var.thor_max_jobs >= 1
+    error_message = "Value must be 1 or more."
+  }
+  default    = 2
+}
+
+variable "thor_num_workers" {
+  type        = number
+  description = "REQUIRED.  The number of Thor workers to allocate.\nMust be 1 or more."
+  validation {
+    condition     = var.thor_num_workers >= 1
+    error_message = "Value must be 1 or more."
+  }
+  default    = 2
+}
+
+variable "thor_worker_cpus" {
+  type        = number
+  description = "The number of CPUs each Thor worker should have.\nMust be 2, 4, 8 or 16. Also, this should be less than 'cpu' in thor node_size."
+  validation {
+    condition     = ((var.thor_worker_cpus == 2) || (var.thor_worker_cpus == 4) || (var.thor_worker_cpus == 8) || (var.thor_worker_cpus == 16))
+    error_message = "Value must be 2, 4, 8, or 16."
+  }
+  default    = 2
+}
+
+###############################################################################
+# Optional variables
+###############################################################################
+
+variable "authn_htpasswd_filename" {
+  type        = string
+  description = "OPTIONAL.  If you would like to use htpasswd to authenticate users to the cluster, enter the filename of the htpasswd file.  This file should be uploaded to the Azure 'dllsshare' file share in order for the HPCC processes to find it.\nA corollary is that persistent storage is enabled.\nAn empty string indicates that htpasswd is not to be used for authentication.\nExample entry: htpasswd.txt"
+  default     = ""
+}
+
+variable "hpcc_namespace" {
```

Comment:
Please break this object out into scalar variables (one of the original design choices in this project), with default values for each. Remember to update the documentation as well.

Reply:
fixed